compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -578,6 +578,172 @@ static int64_t getMaxVectorTileSize(int64_t numElem, int64_t tileSize,
   return 1;
 }
 
+static bool isInnerMostDimThatMapIsFunctionOf(AffineMap map, int dim) {
+  if (!map.isFunctionOfDim(dim)) {
+    return false;
+  }
+  for (int d = dim + 1; d < map.getNumDims(); ++d) {
+    if (map.isFunctionOfDim(d)) {
+      return false;
+    }
+  }
+  return true;
+}
+
+static int sum(ArrayRef<int> a) {
+  int s = 0;
+  for (int i : a) {
+    s += i;
+  }
+  return s;
+}
+
+// Clamps in-place `vecTileSizes`, ensuring that the resulting vector tile sizes
+// for each opearand of `op` satisfy two requirements:
+// 1. No resulting operand tile size exceeds `eachOperandMaxTileBits`.
+// 2. The sum of all resulting operand tile sizes does not exceed
```

Comment:
```suggestion
// 2. The sum of all resulting operand tile size products does not exceed
```

Reply:
I would leave it as "size" meant as "the overall bit size of the entire tile" implying taking the product over all dimensions.